lib/ansible/plugins/action/synchronize.py

```diff
@@ -125,7 +125,10 @@ def run(self, tmp=None, task_vars=dict()):
         # running on so localhost could be a non-controller machine if
         # delegate_to is used)
         src_host  = '127.0.0.1'
-        dest_host = task_vars.get('ansible_ssh_host') or task_vars.get('inventory_hostname')
+        if delegate_to:
+            dest_host = task_vars.get('inventory_hostname')
+        else:
+            dest_host = task_vars.get('ansible_ssh_host') or task_vars.get('inventory_hostname')
 
```

Comment:
This feels wrong because if it matters that we get inventory_hostname when using delegate_to then we wouldn't want to also use inventory_hostname in the non-delegate_to case... that probably means that this happens to make some test cases work but hte real bug is elsewhere (or we shouldn't have inventory_hostname in the else?)  Would need to explore this more to understand what's actually happening here.